### PR TITLE
Feature/freelancers-list-controller

### PR DESCRIPTION
**Pull request by KashyretsIvanna**
what new:
fixes: 
- url,  is Google, user 
feature: search param

check : 
- fixes, search

response (array of freelancers):
зображення

### CODE REVIEW

File: src/modules/jobs/dto/job-params.dto.ts.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  @IsOptional()
+  @IsNumber()
+  hourly_rate?: number;
+
+  @ApiProperty({ example: 3 })
+  @IsOptional()
+  @IsArray()
+  available_time?: number;
+
+  @ApiProperty({ example: 3 })
```
Comment (lyudad) [inline src/modules/jobs/dto/job-params.dto.ts.ts]: wrong example

File: src/modules/jobs/dto/job-params.dto.ts.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { Skill } from '@/common/entities/Skill.entity';
+import { EnglishLevel } from '@/common/constants/entities';
+
+export default class JobParamsDto {
+  @ApiProperty({ example: 3 })
+  @IsOptional()
+  @IsNumber()
+  hourly_rate?: number;
+
+  @ApiProperty({ example: 3 })
```
Comment (lyudad) [inline src/modules/jobs/dto/job-params.dto.ts.ts]: bad example

File: src/modules/jobs/job.service.ts [hunk 3]
```diff
@@ -22,15 +23,24 @@ export class JobsService {
...
-        throw error;
-      }
-      throw new InternalServerErrorException();
-    }
+  async getAllJobs(
+    data: JobParamsDto,
+    limit: number,
+    skip: number,
+  ): Promise<Job[]> {
+    const [res] = await this.jobRepository
```
Comment (lyudad) [inline src/modules/jobs/job.service.ts]: Снимок экрана 2022-11-20 в 20 05 06

File: src/modules/jobs/job.controller.ts [hunk 4]
```diff
@@ -45,4 +57,18 @@ export class JobsController {
...
+  @ApiOperation({ summary: 'get job by id' })
+  @ApiResponse({ type: Job })
+  @ApiHeader({
+    name: 'Authorization',
+    description: 'Bearer token',
+  })
+  @UseGuards(JwtAuthGuard)
+  @Get('/:id')
+  getOneJob(@Param() id: JobIdDto): Promise<AddJobDescriptionDto> {
+    const data = parseInt(id.id, 10);
```
Comment (lyudad) [inline src/modules/jobs/job.controller.ts]: Снимок экрана 2022-11-20 в 20 05 06
item 4

File: src/modules/user/dto/user.dto.ts [hunk 5]
```diff
@@ -12,4 +12,16 @@ export default class UserDto {
...
+  @ApiProperty({ example: '' })
+  available_time: string;
+
+  @ApiProperty({ example: '' })
+  position: string;
+
+  @ApiProperty({ example: '' })
+  hourly_rate: number;
+
+  @ApiProperty({ example: '' })
```
Comment (lyudad) [inline src/modules/user/dto/user.dto.ts]: frontend will use it as documentation, it should be described as good as possible

File: src/modules/user/user.service.ts [hunk 6]
```diff
@@ -238,4 +242,55 @@ export class UserService {
       throw new InternalServerErrorException();
     }
   }
+
+  async getFheelancerInformation(
+    take: number,
+    page: number,
+  ): Promise<[User[], number]> {
+    try {
+      const currentUser = await this.userRepository.findAndCount({
```
Comment (lyudad) [inline src/modules/user/user.service.ts]: query builder

File: src/modules/user/user.service.ts [hunk 7]
```diff
@@ -238,4 +242,55 @@ export class UserService {
...
+      if (error instanceof HttpException) {
+        throw error;
+      }
+      throw new InternalServerErrorException();
+    }
+  }
+
+  async getCategoryInfo(): Promise<Category[]> {
+    try {
+      const Categories = await this.categoryRepository.find();
```
Comment (lyudad) [inline src/modules/user/user.service.ts]: query builder

File: src/modules/jobs/dto/job-params.dto.ts.ts (hunk 1 above)
Comment (KashyretsIvanna) [reply src/modules/jobs/dto/job-params.dto.ts.ts]: fixed

File: src/modules/jobs/dto/job-params.dto.ts.ts (hunk 2 above)
Comment (KashyretsIvanna) [reply src/modules/jobs/dto/job-params.dto.ts.ts]: fixed

File: src/modules/jobs/job.service.ts (hunk 3 above)
Comment (KashyretsIvanna) [reply src/modules/jobs/job.service.ts]: fixed

File: src/modules/jobs/job.controller.ts (hunk 4 above)
Comment (KashyretsIvanna) [reply src/modules/jobs/job.controller.ts]: fixed

File: src/modules/user/dto/user.dto.ts (hunk 5 above)
Comment (KashyretsIvanna) [reply src/modules/user/dto/user.dto.ts]: fixed

File: src/modules/user/user.service.ts (hunk 6 above)
Comment (KashyretsIvanna) [reply src/modules/user/user.service.ts]: fixed

File: src/modules/user/user.service.ts (hunk 7 above)
Comment (KashyretsIvanna) [reply src/modules/user/user.service.ts]: fixed

File: src/modules/auth/dto/signIn.dto.ts [hunk 8]
```diff
@@ -10,4 +10,8 @@ export default class SignInDto {
   @IsOptional()
   @IsString()
   readonly password?: string;
+
+  @IsBoolean()
+  @IsOptional()
+  readonly is_google?: boolean;
```
Comment (lyudad) [inline src/modules/auth/dto/signIn.dto.ts]: why you need this came from BE? you should already know this information if user sign in from google

File: src/modules/user/dto/user.dto.ts [hunk 9]
```diff
@@ -14,9 +14,21 @@ export default class UserDto {
   @ApiProperty({ example: 'Doe' })
   last_name: string;
 
-  @ApiProperty({ example: '' })
+  @ApiProperty({ example: 'https://www.google.com/url' })
```
Comment (lyudad) [inline src/modules/user/dto/user.dto.ts]: why it url? not file?

File: src/modules/user/user.service.ts [hunk 10]
```diff
@@ -252,4 +256,56 @@ export class UserService {
...
+      if (error instanceof HttpException) {
+        throw error;
+      }
+      throw new InternalServerErrorException();
+    }
+  }
+
+  async addCategoryInfo(payload: Category, user: User): Promise<void> {
+    try {
+      const currentUser = await this.findByEmail(user.email);
```
Comment (lyudad) [inline src/modules/user/user.service.ts]: no, you already have a user. its too expensive to get a user for each endpoint, find another way. this function calls for the protected route, then you get a user from guard validation, you can use the same user if needed

Comment (KashyretsIvanna) [reply src/modules/user/user.service.ts]: fixed

File: src/modules/user/dto/user.dto.ts (hunk 9 above)
Comment (KashyretsIvanna) [reply src/modules/user/dto/user.dto.ts]: changed

File: src/modules/auth/dto/signIn.dto.ts (hunk 8 above)
Comment (KashyretsIvanna) [reply src/modules/auth/dto/signIn.dto.ts]: fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**